taskcat/_cli_modules/test.py

```diff
@@ -133,90 +124,33 @@ def run(  # noqa: C901
         :param output_directory: Where to store generated logfiles
         :param minimal_output: Reduces output during test runs
         :param dont_wait_for_delete: Exits immediately after calling stack_delete
+        :param skip_upload: Use templates in an existing cloudformation bucket.
         """
-        project_root_path: Path = Path(project_root).expanduser().resolve()
-        input_file_path: Path = project_root_path / input_file
-        # pylint: disable=too-many-arguments
-        args = _build_args(enable_sig_v2, regions, GLOBAL_ARGS.profile)
-        config = Config.create(
-            project_root=project_root_path,
-            project_config_path=input_file_path,
-            args=args
-            # TODO: detect if input file is taskcat config or CloudFormation template
-        )
-        _trim_regions(regions, config)
-        _trim_tests(test_names, config)
-        boto3_cache = Boto3Cache()
-        templates = config.get_templates()
-        if skip_upload and not config.config.project.s3_bucket:
-            raise TaskCatException(
-                "cannot skip_buckets without specifying s3_bucket in config"
-            )
-        buckets = config.get_buckets(boto3_cache)
-        if not skip_upload:
-            # 1. lint
-            if not lint_disable:
-                lint = TaskCatLint(config, templates)
-                errors = lint.lints[1]
-                lint.output_results()
-                if errors or not lint.passed:
-                    raise TaskCatException("Lint failed with errors")
-            # 2. build lambdas
-            if config.config.project.package_lambda:
-                LambdaBuild(config, project_root_path)
-            # 3. s3 sync
-            stage_in_s3(buckets, config.config.project.name, config.project_root)
-        # 4. launch stacks
-        regions = config.get_regions(boto3_cache)
-        parameters = config.get_rendered_parameters(buckets, regions, templates)
-        tests = config.get_tests(templates, regions, buckets, parameters)
-        test_definition = Stacker(
-            config.config.project.name,
-            tests,
-            shorten_stack_name=config.config.project.shorten_stack_name,
+
+        test = CFNTest.from_file(
+            project_root=project_root,
+            input_file=input_file,
+            regions=regions,
+            enable_sig_v2=enable_sig_v2,
         )
-        test_definition.create_stacks()
+
+        # This code is temporary and should be removed once its easier
+        # to create a config object
+        frame = inspect.currentframe()
+
+        if frame is not None:
+            args, _, _, values = inspect.getargvalues(frame)
+
+            for i in args:
+                if hasattr(test, i):
+                    setattr(test, i, values[i])
```

Comment:
In a future PR I hope to make some improvements to Config so that we can kill this code with 🔥 .

What the above does is grabs all the function arguments and then sets all those values on the test object.

Reply:
Yup. I've been eyeing a more constructive Config object right behind what's currently on my plate for EOY delivery